### DISCUSSION TITLE
How to add a custom stimulus controller

### DISCUSSION THREAD

**Discussion by iainbeeston** [avo-hq/avo#962] [Help]
I'd like to use a custom stimulus controller, but although I've read the instructions [here](https://docs.avohq.io/2.0/stimulus-integration.html#custom-stimulus-controllers), it seems that my custom controller (in `app/javascripts/controllers`) isn't being loaded by avo. I had assumed that if I added the name of my custom controller to `self.stimulus_controllers` in my resource class, it would be loaded. Could you please explain how to get avo to load my custom controller?

**Comment by adrianthedev** [comment]
Hmm, I now see what you mean. I haven't mentioned that you'd need to add your own custom asset pipeline to get the JS code loaded. 

The gist is that you'll have to use your own custom asset pipeline to load the JS code to the browser. If you use sprockets on the front office of your app, you'll be able to use that. Same for Webpacker.
You need to create a new JS file where you import and declare your stimulus controllers and load it in the `_head.html.erb` partial (which you can eject using `bin/rails generate avo:eject :head`) using `<%= javascript_include_tag 'avo_application' %>`.

Please follow the short instructions [here](https://docs.avohq.io/2.0/custom-asset-pipeline.html#sprockets) to add JS code to your app.

Please let me know if there's something I missed while your try this solution.

I'll update the docs with what I wrote above.
